src/components/ListComposable/types.ts

```diff
@@ -0,0 +1,148 @@
+import type {QAProps} from '../types';
+
+import type {modToHeight} from './constants';
+
+export type ListItemId = string;
+/**
+ * Default fallback id type.
+ * The core idea is to store original index and and nested children index
+ */
+export type GroupedId = string;
+
+export interface ListItemType<T = unknown> {
+    /**
+     * Required if you want to control list state from internal place.
+     * For example to control expanded groups state
+     */
+    id?: string;
+    data: T;
+    children?: ListItemType<T>[];
+}
+export type GroupParsedState = {
+    childrenCount: number;
+};
+
+export type ItemParsedState = {
+    indentation: number;
+};
+export type ParsedState<T = unknown> = {
+    /**
+     * stored internal meta info about item
+     * Note: Groups are also items
+     */
+    itemsState: Record<ListItemId, ItemParsedState>;
+    /**
+     * Normalized original data
+     */
+    byId: Record<ListItemId, T>;
+    /**
+     * Re
+     */
+    groupsState: Record<ListItemId, GroupParsedState>;
+};
+
+export type ListSizeTypes = keyof typeof modToHeight;
+
+export interface ListItemBaseData {
+    title: string;
+    subtitle?: string;
+    leftSlot?: React.ReactNode;
+    rightSlot?: React.ReactNode;
```

Comment:
Due to upcoming RTL support it's better to name these two like `startSlot` and `endSlot`